drafts/use-cases/2.api-documentation.md

```diff
@@ -0,0 +1,111 @@
+## API Documentation
+
+### Story
+As an application user
+I want my application to be able to perform some precomputations 
+So it can pre-generate some views and other UI elements to speed things up.
+
+### Usage
+```javascript
+var client = new HydraClient();
+var apiDocumentation = client.getApiDocumentation("/api");
+for (var $class of apiDocumentation.supportedClasses) {
+    // generate a partial view for the _$class_
+}
+```
+
+### Details
+The documentation obtain may be a large payload,
+thus client is expected to store it temporarely at least for the current session.
+
+```
+GET /api?documentation
+
+HTTP 200 OK
+```
+```javascript
+{
+    "@id": "/api?documentation",
+    "supportedClasses": [
+        {
+            "@id": "schema:Event",
+            "suportedProperties": [
+                {
+                    "property": {
+                        "@id": "eventName"
+                    }
+                },
+                {
+                    "property": {
+                        "@id": "eventDescription"
+                    }
+                },
+                {
+                    "property": {
+                        "@id": "startDate"
+                    }
+                },
+                {
+                    "property": {
+                        "@id": "endDate"
+                    }
+                }
+            ]
+        }
+    ]
+}
+```
+
+In order to i.e. generate views for each supported class, a complete type definition should be provided.
+Currently, Hydra Core Vocabulary allows that using RDFS/OWL vocabularies.
+
+The documentation should also contain some human readable details like:
+- operation expected type and it's role
+- operation returned type and it's role
+- operation name
+- operation description
+- parameters and their descriptions
+
+### Considerations
+
+#### Entry-Point/API Documentation fetching precedence
+In the previous scenario, application fetched an entry-point data as it's first action.
+We also could consider an opposite - taking an API documentation if available.
+This would require some standardized mechanism of accesing that documentation.
+
+Few possibilities here are available:
+- sending an _OPTIONS_ HTTP request to the base URL,
+  but HTTP specification does not allow to cache these responses
+- sending an _GET_ HTTP request to the base URL with some HTTP header;
+  header _Expect_ would be a candidate, but it's not allowed in AJAX communication;
+  another option would be a _Prefer_
+
+Once the API documentation is fetched and processed,
+client would then display some base view with further options,
+i.e. a main menu with all supported classes.
```

Comment:
> display all supported classes

What would the client do next? I expect the client to work with resource representations. The documentation tells the client how to do it but it is not its main interest.